### ISSUE TITLE
Bug: no-extra-parens creates no-mixed-operators error

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#17541]
### Environment

See https://github.com/eslint/eslint/issues/13413

### What parser are you using?

Default (Espree)

### What did you do?

See https://github.com/eslint/eslint/issues/13413

### What did you expect to happen?

no-extra-parens should only remove the extra parents when they are redundant and won't cause a no-mixed-operators error.
Essentially if there are || or && in the parenthesis, it should not remove them (first level).

Using `'nestedBinaryExpressions': false` is not a good option, since it will not remove ANY extra parenthesis.

`if ( ( foo in bar ) || x in y ) {` should be fixed, but `if ( ( foo in bar || x in y ) && acme === 2 ) {` not

### What actually happened?

Removed () and caused no-mixed-operators

### Link to Minimal Reproducible Example

https://github.com/eslint/eslint/issues/13413

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @kkmuffme, thanks for the issue!

This works as intended. As described in the docs for [`no-mixed-operators`](https://eslint.org/docs/latest/rules/no-mixed-operators#rule-details), If you use both this and `no-extra-parens` rule together, you need to use the `nestedBinaryExpressions` option of `no-extra-parens` rule.

https://github.com/eslint/eslint/blob/da09f4e641141f585ef611c6e9d63d4331054706/docs/src/rules/no-mixed-operators.md?plain=1#L36-L37

> no-extra-parens should only remove the extra parents when they are redundant and won't cause a no-mixed-operators error.
> Essentially if there are || or && in the parenthesis, it should not remove them (first level).

These are separate rules. `no-mixed-operators` can be configured differently so `no-extra-parens` can't know which cases don't cause `no-mixed-operators` errors.

> Using `'nestedBinaryExpressions': false` is not a good option, since it will not remove ANY extra parenthesis.

`'nestedBinaryExpressions': false` only allows parentheses around nested binary expressions. All other unnecessary parentheses are still reported.

**Comment by mdjermanovic** [comment]
> No, it doesn't. When using `'nestedBinaryExpressions': false` it will not change:
> `if ( typeof hello === 'object' && ( foo in bar ) || x in y ) {`
> even though it should

`foo in bar` is a binary expression and is an operand of binary `&&` so it's considered a nested binary expression and therefore `no-extra-parens` with `'nestedBinaryExpressions': false` doesn't report these parentheses.

Removing these parentheses can cause conflicts with `no-mixed-operators`:

```js
/* eslint no-extra-parens: ["error", "all", { nestedBinaryExpressions: false }] */
/* eslint no-mixed-operators: ["error", { groups: [["&&", "in"]] }] */

if ( typeof hello === 'object' && ( foo in bar ) || x in y ) {}

if ( typeof hello === 'object' &&   foo in bar   || x in y ) {} // no-mixed-operators error
```

[Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWV4dHJhLXBhcmVuczogW1wiZXJyb3JcIiwgXCJhbGxcIiwgeyBuZXN0ZWRCaW5hcnlFeHByZXNzaW9uczogZmFsc2UgfV0gKi9cbi8qIGVzbGludCBuby1taXhlZC1vcGVyYXRvcnM6IFtcImVycm9yXCIsIHsgZ3JvdXBzOiBbW1wiJiZcIiwgXCJpblwiXV0gfV0gKi9cblxuaWYgKCB0eXBlb2YgaGVsbG8gPT09ICdvYmplY3QnICYmICggZm9vIGluIGJhciApIHx8IHggaW4geSApIHt9XG5cbmlmICggdHlwZW9mIGhlbGxvID09PSAnb2JqZWN0JyAmJiAgIGZvbyBpbiBiYXIgICB8fCB4IGluIHkgKSB7fSAvLyBuby1taXhlZC1vcGVyYXRvcnMgZXJyb3IiLCJvcHRpb25zIjp7ImVudiI6eyJlczYiOnRydWV9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCJ9LCJydWxlcyI6e319fQ==)

**Comment by mdjermanovic** [comment]
I don't see what would constitute a bug in the above example. `no-mixed-operators` requires parentheses when configured operators (in this case, `&&` and `in`) are mixed. The point of this rule is to clarify the order of operations by adding parentheses, which will be redundant if the precedence matches the developer’s intention.
